browser/app/js/components/Browse.js

```diff
@@ -145,12 +145,43 @@ export default class Browse extends React.Component {
   selectPrefix(e, prefix) {
     e.preventDefault()
     const {dispatch, currentPath, web, currentBucket} = this.props
-    const encPrefix = encodeURI(prefix)
+    const objectName = encodeURI(prefix)
     if (prefix.endsWith('/') || prefix === '') {
       if (prefix === currentPath) return
-      browserHistory.push(utils.pathJoin(currentBucket, encPrefix))
+      browserHistory.push(utils.pathJoin(currentBucket, objectName))
     } else {
-      window.location = `${window.location.origin}/minio/download/${currentBucket}/${encPrefix}?token=${storage.getItem('token')}`
+      let xhr = new XMLHttpRequest()
+      let url = `${location.origin}/minio/download/${currentBucket}/${objectName}`
+
+      xhr.open('GET', url, true)
+      xhr.setRequestHeader('Authorization', 'Bearer ' + storage.getItem('token'))
+
+      // The response must allow binary data.
+      xhr.responseType = 'arraybuffer'
```

Comment:
Can you check what is better for us here `arraybuffer` or `blob`? Because we don't want the browser to cache everything in memory first before writing to the disk. Browser should save it to the disk in a streaming fashion.